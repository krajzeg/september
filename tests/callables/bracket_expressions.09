syntax::"[{}]" = |...items, :::properties| {
	[[
		array: items,
		object: properties
	]]
}

both := [{1, 2, 3, apples: 4, bananas: 5}]

print("The array part should print 1,2,3.")
for (item) in (both.array) {
	print(item)
}
print("The object part should print 4,5.")
print(both.object.apples)
print(both.object.bananas)

empty := [{}]
print("Empty brackets should work, so this should be True, True: ", empty.array.is(Array), empty.object.is(Object))
