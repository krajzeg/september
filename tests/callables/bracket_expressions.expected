The array part should print 1,2,3.
1
2
3
The object part should print 4,5.
4
5
Empty brackets should work, so this should be True, True:  <True> <True>
